src/vehicle.cpp

```diff
@@ -2435,30 +2495,34 @@ void vehicle::print_fuel_indicator (void *w, int y, int x, bool fullsize, bool v
     int yofs = 0;
     int max_gauge = (isHorizontal) ? 12 : 5;
     int cur_gauge = 0;
-    std::vector< ammotype > fuels( &fuel_types[0], &fuel_types[num_fuel_types] );
-    // Find non-hardcoded fuel types, add them after the hardcoded
-    for( int p : fuel ) {
-        const ammotype ft = part_info( p ).fuel_type;
-        if( std::find( fuels.begin(), fuels.end(), ft ) == fuels.end() ) {
-            fuels.push_back( ft );
-        }
-    }
-
-    for( int i = 0; i < (int)fuels.size(); i++ ) {
-        const ammotype &f = fuels[i];
+    const auto &fuels = all_liquids();
+    // Sort for better display
+    // Can't define a map with a comparer - for some reason skips entries when iterating
```

Comment:
This works for me (only compiled, not executed):
```c++
std::map<std::string, long> fuels = ...;
std::set<fuel_val, fuel_sort> fuels_sorted(fuels.begin(), fuels.end());
```

But your comparator is wrong. For the entries `a = fuel_val{ "gasoline", 200 }` and `b = fuel_val{ "diesel", 100 }`, both direction hold true:

`fuel_sort()(a, b) == true` (because `ldist < rdist`)
`fuel_sort()(b, a) == true` (because `left.second > right.second`)

That seems to confuse the map, because it means `a < b`, but also `b < a` and that can not be.

Those comparators are usually implemented like this:
```C++
if( left.property1 != right.property1 ) { return left.property1 < right.property1; }
if( left.property2 != right.property2 ) { return left.property2 < right.property2; }
...
return false; // all properties are equal, therefor "left < right" is false
```